### ISSUE TITLE
Error: ENOENT, open '/tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack/package.json'

### ISSUE THREAD

**Issue by wilpatterson** [npm/npm#11701]
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/nodejs', '/usr/bin/npm', 'install' ]
2 info using npm@1.4.21
3 info using node@v0.10.25
4 verbose readDependencies using package.json deps
5 verbose install where, deps [ '/home/will/Rail/railnet-server/server',
5 verbose install   [ 'http',
5 verbose install     'moniker',
5 verbose install     'node-event-emitter',
5 verbose install     'rai',
5 verbose install     'underscore',
5 verbose install     'railnet-api' ] ]
6 info preinstall railnetd@1.0.0
7 warn package.json http@0.0.0 No repository field.
8 warn package.json moniker@0.1.2 No repository field.
9 verbose readDependencies using package.json deps
10 verbose already installed skipping http@* /home/will/Rail/railnet-server/server
11 verbose already installed skipping moniker@^0.1.2 /home/will/Rail/railnet-server/server
12 verbose already installed skipping node-event-emitter@0.0.1 /home/will/Rail/railnet-server/server
13 verbose already installed skipping rai@^0.1.12 /home/will/Rail/railnet-server/server
14 verbose already installed skipping underscore@^1.8.3 /home/will/Rail/railnet-server/server
15 verbose cache add [ 'railnet-api@git+https://github.com/wilpatterson/railnet-api.git',
15 verbose cache add   null ]
16 verbose cache add name=undefined spec="railnet-api@git+https://github.com/wilpatterson/railnet-api.git" args=["railnet-api@git+https://github.com/wilpatterson/railnet-api.git",null]
17 verbose parsed url { protocol: null,
17 verbose parsed url   slashes: null,
17 verbose parsed url   auth: null,
17 verbose parsed url   host: null,
17 verbose parsed url   port: null,
17 verbose parsed url   hostname: null,
17 verbose parsed url   hash: null,
17 verbose parsed url   search: null,
17 verbose parsed url   query: null,
17 verbose parsed url   pathname: 'railnet-api@git+https://github.com/wilpatterson/railnet-api.git',
17 verbose parsed url   path: 'railnet-api@git+https://github.com/wilpatterson/railnet-api.git',
17 verbose parsed url   href: 'railnet-api@git+https://github.com/wilpatterson/railnet-api.git' }
18 verbose cache add name="railnet-api" spec="git+https://github.com/wilpatterson/railnet-api.git" args=["railnet-api","git+https://github.com/wilpatterson/railnet-api.git"]
19 verbose parsed url { protocol: 'git+https:',
19 verbose parsed url   slashes: true,
19 verbose parsed url   auth: null,
19 verbose parsed url   host: 'github.com',
19 verbose parsed url   port: null,
19 verbose parsed url   hostname: 'github.com',
19 verbose parsed url   hash: null,
19 verbose parsed url   search: null,
19 verbose parsed url   query: null,
19 verbose parsed url   pathname: '/wilpatterson/railnet-api.git',
19 verbose parsed url   path: '/wilpatterson/railnet-api.git',
19 verbose parsed url   href: 'git+https://github.com/wilpatterson/railnet-api.git' }
20 silly lockFile 169e2ea1-com-wilpatterson-railnet-api-git https://github.com/wilpatterson/railnet-api.git
21 verbose lock https://github.com/wilpatterson/railnet-api.git /home/will/.npm/169e2ea1-com-wilpatterson-railnet-api-git.lock
22 verbose addRemoteGit [ 'https://github.com/wilpatterson/railnet-api.git', 'master' ]
23 verbose git remote.origin.url https://github.com/wilpatterson/railnet-api.git
24 verbose git fetch -a origin (https://github.com/wilpatterson/railnet-api.git)
25 verbose git rev-list -n1 master 04d01b983a7026ee1445455b37558d4d087817f4
26 verbose resolved git url git+https://github.com/wilpatterson/railnet-api.git#04d01b983a7026ee1445455b37558d4d087817f4
27 verbose tar unpack /tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz
28 silly lockFile 22af5496-10838637454435229-tmp-tgz-unpack tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack
29 verbose lock tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack /home/will/.npm/22af5496-10838637454435229-tmp-tgz-unpack.lock
30 silly lockFile 423844ed-6046-0-10838637454435229-tmp-tgz tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz
31 verbose lock tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz /home/will/.npm/423844ed-6046-0-10838637454435229-tmp-tgz.lock
32 silly gunzTarPerm modes [ '755', '644' ]
33 silly gunzTarPerm extractEntry
34 silly gunzTarPerm modified mode [ '', 509, 493 ]
35 silly gunzTarPerm extractEntry emitshell.js
36 silly gunzTarPerm modified mode [ 'emitshell.js', 436, 420 ]
37 silly gunzTarPerm extractEntry messages.js
38 silly gunzTarPerm modified mode [ 'messages.js', 436, 420 ]
39 silly lockFile 22af5496-10838637454435229-tmp-tgz-unpack tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack
40 silly lockFile 22af5496-10838637454435229-tmp-tgz-unpack tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack
41 silly lockFile 423844ed-6046-0-10838637454435229-tmp-tgz tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz
42 silly lockFile 423844ed-6046-0-10838637454435229-tmp-tgz tar:///tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz
43 silly lockFile 169e2ea1-com-wilpatterson-railnet-api-git https://github.com/wilpatterson/railnet-api.git
44 silly lockFile 169e2ea1-com-wilpatterson-railnet-api-git https://github.com/wilpatterson/railnet-api.git
45 error Error: ENOENT, open '/tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack/package.json'
46 error If you need help, you may report this *entire* log,
46 error including the npm and node versions, at:
46 error     <http://github.com/npm/npm/issues>
47 error System Linux 4.2.0-30-generic
48 error command "/usr/bin/nodejs" "/usr/bin/npm" "install"
49 error cwd /home/will/Rail/railnet-server/server
50 error node -v v0.10.25
51 error npm -v 1.4.21
52 error path /tmp/npm-16255-r0xY-D43/1456519426046-0.10838637454435229/tmp.tgz-unpack/package.json
53 error code ENOENT
54 error errno 34
55 verbose exit [ 34, true ]
```

```
{
  "name": "railnetd",
  "version": "1.0.0",
  "description": "Node server for RPi to act as the server for railnet.",
  "repository": "https://github.com/TableSafe/railnet-server",
  "main": "server.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "Michael Weaver",
  "license": "MIT",
  "dependencies": {
    "http": "*",
    "moniker": "^0.1.2",
    "node-event-emitter": "0.0.1",
    "rai": "^0.1.12",
    "underscore": "^1.8.3",
    "railnet-api": "git+https://github.com/wilpatterson/railnet-api.git"
  }
}
```

**Comment by KenanY** [comment]
@wilpatterson Does this also occur if you [update `npm`](https://github.com/npm/npm/wiki/Troubleshooting#try-the-latest-stable-version-of-npm)?

**Comment by iarna** [comment]
@wilpatternson We've fixed a LOT of bugs since the 1.4 days– please update (following the instructions KenenY linked to) and try again. I'm going to close this issue, but if this doesn't resolve your problem, please open a new issue with the error you're getting from your updated npm.

**Comment by iarna** [comment]
(Alternatively, you can email `support@npmjs.com` and they'll be happy to work through things with you.)
